templates/federated_reporting/transport.sh

```diff
@@ -21,28 +33,108 @@ if ! type "$CFE_FR_TRANSPORTER" >/dev/null; then
   exit 1
 fi
 
-dump_files="$(ls -1 "$CFE_FR_TRANSPORT_DIR/"*".sql.$CFE_FR_COMPRESSOR_EXT" 2>/dev/null)" ||
-  {
-    log "No files to transport."
-    exit 0
-  }
+function push() {
+  # Runs on the feeder hub, pushes dumps to the superhub.
+
+  dump_files="$(ls -1 "$CFE_FR_TRANSPORT_DIR/"*".sql.$CFE_FR_COMPRESSOR_EXT" 2>/dev/null)" ||
+    {
+      log "No files to transport."
+      exit 0
+    }
+
+  log "Transporting files: $dump_files"
+  some_failed=0
+  for dump_file in $dump_files; do
+    failed=0
+    mv "$dump_file" "$dump_file.transporting"
+    "$CFE_FR_TRANSPORTER" "$CFE_FR_TRANSPORTER_ARGS" "$dump_file.transporting" "$CFE_FR_SUPERHUB_LOGIN:$CFE_FR_SUPERHUB_DROP_DIR/$(basename "$dump_file")" &&
```

Comment:
in case of failure should we not `mv` the files but rather `cp` them and then delete them if not failed?

Reply:
So this is a generic question about what should happen when something fails. I talked to @olehermanse about this and we agreed that primarily we should make sure things are not piling up. So what I think should actually be done here is that the file should be removed after the transport attempt in any case. But let's not change that in this commit or PR because that's just moving the "old" push code, not modifying it.